src/components/FormDelayTimeControl.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div class="form-group">
+    <label>{{ label }}</label>
+    <div class="input-group">
+      <input type="number" min="1" class="form-control control repeat" :class="classList" v-model="repeat" >
+      <div class="input-group-append">
+        <select class="form-control custom-select" :class="classList" v-model="periodicity">
+          <option v-for="period in periods" :key="period.name" :value="period">{{ $t(period.name) }}</option>
+        </select>
+      </div>
+    <template v-if="validator && validator.errorCount">
+      <div class="invalid-feedback" v-for="(errors, index) in validator.errors.all()" :key="index">
+        <div v-for="(error, subIndex) in errors" :key="subIndex">
+          {{error}}
+        </div>
+      </div>
+    </template>
+    <div class="invalid-feedback" v-if="error">{{error}}</div>
+    </div>
+    <small v-if="helper" class="form-text text-muted">{{ helper }}</small>
+  </div>
+</template>
+
+<script>
+import ValidationMixin from './mixins/validation'
+import last from 'lodash/last';
+
+const periodNames = {
+  minute: 'minute',
+  hour: 'hour',
+  day: 'day',
+  month: 'month',
+};
+
+export default {
+  mixins: [ValidationMixin],
+  props: {
+    value: {
+      type: String,
+      default: 'PT1M',
+    },
+    error: null,
+    name: String,
+    label: String,
+    helper: String,
+  },
+  data() {
+    const periods = [
+      { name: periodNames.minute, value: 'M', isTime: true },
+      { name: periodNames.hour, value: 'H', isTime: true },
+      { name: periodNames.day, value: 'D' },
+      { name: periodNames.month, value: 'M' },
+    ];
+
+    return {
+      repeat: null,
+      periodicity: null,
+      periods,
+    };
+  },
+  watch: {
+    value: {
+      handler(value) {
+        this.periodicity = this.getPeriodFromDelayString(value);
+        this.repeat = this.getRepeatNumberFromDelayString(value);
+      },
+      immediate: true,
+    },
+    durationExpression: {
+      handler(durationExpression) {
+        this.$emit('input', durationExpression);
+      },
+      immediate: true,
+    },
+  },
+  computed: {
+    classList(){
+      let classList = {
```

Comment:
This can be cleaned up:
```js
classList() {
  return {
    'is-invalid': (this.validator && this.validator.errorCount) || this.error,
    [this.controlClass]: !!this.controlClass
  }
}
```